src/core/thread/network_data_service.cpp

```diff
@@ -141,6 +137,24 @@ void Manager::GetBackboneRouterPrimary(ot::BackboneRouter::BackboneRouterConfig
     return;
 }
 
+bool Manager::IsBackboneRouterPrimaryPriorTo(const ServerTlv &                 aServerTlv,
+                                             const BackboneRouter::ServerData &aServerData,
+                                             const ServerTlv &                 aOtherServerTlv,
+                                             const BackboneRouter::ServerData &aOtherServerData) const
+{
+    bool     isPrior;
+    uint16_t leaderRloc16 = Mle::Mle::Rloc16FromRouterId(Get<Mle::MleRouter>().GetLeaderId());
+
+    VerifyOrExit(aServerTlv.GetServer16() != leaderRloc16, isPrior = true);
+    VerifyOrExit(aOtherServerTlv.GetServer16() != leaderRloc16, isPrior = false);
```

Comment:
not necessary to have this line if `isPrior` is initialized as `false`?

Reply:
`isPrior` is not initialized.